libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesObserver.kt

```diff
@@ -13,9 +13,14 @@ fun interface RoutesObserver {
      *
      * The route at index 0, if exist, will be treated as the primary route for 'Active Guidance'.
      *
-     * A list of routes can be modified internally and externally at any time with [MapboxNavigation.setRoutes], or during automatic reroutes, faster route and route refresh operations.
+     * A list of routes can be modified internally and externally at any time with
+     * [MapboxNavigation.setRoutes], or during automatic reroutes, faster route and route refresh operations.
      *
      * @param routes list of currently maintained routes
+     * @param reason a reason routes have been updated (re-route, refresh route, and etc.)
      */
-    fun onRoutesChanged(routes: List<DirectionsRoute>)
+    fun onRoutesChanged(
+        routes: List<DirectionsRoute>,
+        @RoutesExtra.RoutesUpdateReason reason: String,
```

Comment:
What if we added a data transfer object here
```suggestion
        changeResult: RoutesChangeResult,
```

with an internal constructor, so that this type of a change won't be breaking in the future:
```kotlin
class RoutesChangeResult internal constructor(
  val routes: List<DirectionsRoute>,
  val @RoutesExtra.RoutesUpdateReason reason: String
)
```

Reply:
💯 make sense